test/swar/BasicOperations.cpp

```diff
@@ -5,6 +5,7 @@
 #include <ios>
 #include <iomanip>
 #include <iostream>
+#include <cstring>
```

Comment:
I don't see value in using the `std::` instead of the functions in the global namespace.
Why `ctstring`?

Reply:
I'm indifferent, string.h is fine.